### ISSUE TITLE
GRPC TraceContext Tags overwrite warnings

### ISSUE THREAD

**Issue by mathewc** [Azure/azure-functions-host#9651]
Seeing lots of warnings coming from [this](https://github.com/Azure/azure-functions-host/blame/6ae170c32c97864d33424e1f8cc1783acdbc907a/src/WebJobs.Script.Grpc/Extensions/ScriptInvocationContextExtensions.cs#L242) code, indicating that we're often overwriting trace context values.

This doesn't seem right. We should investigate to see why this happens and if it should be a Warning.

FunctionsLogs
| where PreciseTimeStamp > ago(1d)
| where Level == 3
| where Summary contains "writing"
| where Source startswith "Worker.LanguageWorkerChannel."
| where Summary startswith "Overwriting"
| project PreciseTimeStamp, Level, Summary
| take 10

**Comment by jviau** [comment]
Saw this for a couple node apps today. It looks like we are for some reason mapping the entire log scope back into `Activity.Tags`. Here is where we do it: https://github.com/Azure/azure-webjobs-sdk/blob/dev/src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/ApplicationInsightsLogger.cs#L412

The big question is, why do we do this? It doesn't seem right. This is actually causing **significant** log churn when it is hit, leading to noticeable perf impact.

**Comment by timjensen-kr** [comment]
This could be contributing to a recurring production issue we have started observing recently in our NODEJS app. We have an instance hitting this that then goes into an unhealthy state for 2+ hours without recovering.

**Comment by jviau** [comment]
@timjensen-kr yeah I am not sure if this issue would contribute to any production problem, or be the result of another issue. I am still investigating. But this is from the AppInsights logger. If you switch to OTel, this issue will not occur.
